modules/registrars/openprovider/init.php

```diff
@@ -77,15 +81,48 @@ function openprovider_bind_required_classes($launcher)
         $client = new ApiV1($logger, $camelCaseToSnakeCaseNameConverter, $idn);
         $client->getConfiguration()->setHost($host);
 
-        if (!$session->has('AUTH_TOKEN')) {
+        $tokenResult = null;
+
+        if (Capsule::schema()->hasTable('reseller_tokens')) {
+            $tokenResult = Capsule::table('reseller_tokens')->where('username', $params['Username'])->orderBy('created_at', 'desc')->first();
+        } else {
+            Capsule::schema()->create(
+                'reseller_tokens',
+                function ($table) {
+                    /** @var \Illuminate\Database\Schema\Blueprint $table */
+                    $table->increments('id');
+                    $table->string('username');
+                    $table->string('token');
+                    $table->string('expire_at');
+                    $table->string('created_at');
+                }
+            );
+        }
+
+        $token = "";
+        $expireTime = is_null($tokenResult) ? false : new Carbon($tokenResult->expire_at);
+        $isExpired = $expireTime ? Carbon::now()->diffInSeconds($expireTime, false) < 0 : true;
```

Comment:
The same here:
$isAlive = $expireTime && Carbon::now()->diffInSeconds($expireTime, false) < 0;
if ($tokenResult && $isAlive) {
...